src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1546,7 +1550,7 @@ def create_vm(vm_name, resource_group_name, image=None, size='Standard_DS1_v2',
     vm_dependencies = []
     if storage_account_type == 'new':
         storage_account = storage_account or 'vhdstorage{}'.format(
-            random_string(14, force_lower=True))
+            hash_string(vm_id, length=14, force_lower=True))
```

Comment:
Note, for storage account, we don't need this logic, because the subsequent `vm create` will find and reuse the account. If this is for code consistency, then it is fine.